.github/workflows/leetcode_directory_writer.yml

```
@@ -0,0 +1,23 @@
+# The objective of this GitHub Action is to update the leetcode DIRECTORY.md file (if needed)
+# when doing a git push
+name: leetcode_directory_writer
+on: [push]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1  # v1, NOT v2 or v3

```

With this, we'll be able to pull all the changes before committing as well as a newer version.
```suggestion
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
```